cachix/src/Cachix/Client/Push.hs

```diff
@@ -172,14 +186,15 @@ uploadStorePath cache storePath retrystatus = do
   normalized <- liftIO $ Store.followLinksToStorePath store $ toS storePathText
   pathinfo <- liftIO $ Store.queryPathInfo store normalized
   let storePathSize = Store.validPathInfoNarSize pathinfo
-  onAttempt strategy retrystatus storePathSize
+  maybeProgressBar <- onAttempt strategy retrystatus storePathSize
 
   withCompressor $ \compressor -> liftIO $ do
     uploadResult <-
       runConduitRes $
         streamNarIO narEffectsIO (toS storePathText) Data.Conduit.yield
           .| passthroughSizeSink narSizeRef
           .| passthroughHashSink narHashRef
+          .| updateProgressBar maybeProgressBar
```

Comment:
Make this more general to achieve better separation of concerns. See comment in `PushStrategy`.

```suggestion
          .| spyIntoNarStream retrystatus storePathSize
```